### DISCUSSION TITLE
Failed to connect to github.com port 443 after 21124 ms: Couldn't connect to server

### DISCUSSION THREAD

**Discussion by marcolopes** [community/community#66424] [Repositories] [answered]
### Select Topic Area

Question

### Body

We are having connection issues with GIT, even using GitHub Desktop APP!

_**fatal: unable to access 'https://github.com/marcolopes/dma.git/': Failed to connect to github.com port 443 after 21124 ms: Couldn't connect to server**_

The connection using ECLIPSE SVN is SLOW SLOW SLOW, and hangs sometimes (when it doesen't hang, it will throw "**org.apache.subversion.javahl.ClientException: Can't create session
svn: Unable to connect to a repository at URL 'https://github.com/marcolopes/dma/trunk/org.dma.eclipse'
Connection timed out
svn: Connection timed out**"

Is there a problem with GIT?

We are used to it being slow using SVN, but this has nothing to do with it, since direct connection with GitHub Desktop (Windows) has the same problems...

Thanks!

**Comment by CMon** [comment]
we encounter the same on our machines, and its not restricted to a specific repo, commands that fail on the first run succeed on a second run. Not a solution for our build pipelines.

```
9:15:34 ~$ git ls-remote --tags --heads https://github.com/mgcrea/angular-strap.git
fatal: unable to access 'https://github.com/mgcrea/angular-strap.git/': Failed to connect to github.com port 443 after 129237 ms: Couldn't connect to server
9:17:45 ~$ git ls-remote --tags --heads https://github.com/mgcrea/angular-strap.git
005ca9c9fbb7651156139927ac38b29d850fb004        refs/heads/gh-pages
c024601dd8866181c3aab1567a05b68290840ef9        refs/heads/legacy
957bbf9841ff99f847d1f97153294f902e465b92        refs/heads/master
d8895dc39963cacf34ef084c6b8a58250c14802c        refs/tags/v0.5.10
81a830ec22d6f60ffb0413fafc85c1052d5531bd        refs/tags/v0.5.6
b6bdc373eea525c8a6fb77ef617288f67e56aff0        refs/tags/v0.5.7
...
```

**Comment by huelsmc** [comment]
Same issue here. Connections to github.com are failing sometimes. Another run often (not 100%) is successful.

**Reply by huelsmc** [reply]
In our case it's happening during a github workflow running on github runners but sometimes also own runners

**Reply by baizhu0414** [reply]
还真可以，离谱

**Reply by codeSoupSz** [reply]
> 还真可以，离谱

我也是这样，离了个大谱

**Reply by FrankChen070716** [reply]
fucking amazing

**Reply by FrankChen070716** [reply]
> > 还真可以，离谱
> 
> 我也是这样，离了个大谱

一样

**Reply by ZhangSan-fwkt** [reply]
重试了几次，还真可以了，离谱

**Reply by NingxinSu** [reply]
same,离谱

**Reply by BowenGuan** [reply]
how f**king amazing!!!

**Comment by kpaaro** [comment]
I tried running curl https://github.com on our CI server and every 1/3 request hangs. The issue poped up when pulling from repos to start build pipelines. This behaviour was first noticed 24h ago in our CI server.

**Reply by kpaaro** [reply]
I also  added a ticket in github support.

**Reply by marcolopes** [reply]
How can i do that? I tried and failed! :D

**Reply by kpaaro** [reply]
Which bit? support ticket? https://support.github.com

**Comment by jkralik** [comment]
I have also same issue with own runner:
```
Failed to connect to github.com port 443 after 130098 ms: Connection timed out
```

**Comment by hashbou** [comment]
Hello, we're having the same problem. It happens regularly and breaks our pipeline builds currently. We ruled out rate limiting and other network issues. It really seems to happen that the git servers at GitHub are not responsive from time to time. 

We recognized this also since early yesterday. 

Maybe this is related to: https://www.githubstatus.com/incidents/2gy2gddtv23d

**Reply by marcolopes** [reply]
Seems related... Started yesterday...

_We are investigating reports of degraded performance for Git Operations
Posted 1 day ago. Sep 07, 2023 - 10:02 UTC_

**Comment by AlexisFronteau** [comment]
Hi, we're having the same problem since yesterday and it is not resolved

**Comment by samwyndham** [comment]
We have also been experiencing issues since yesterday frequently resulting in failing GitHub action jobs. It is mainly manifesting itself as warning failing to download Actions, and errors using GitHub API over GitHubs CLI tool. I think it may relate to this issue: https://github.com/orgs/community/discussions/66444

**Comment by samwyndham** [comment]
I'm surprised that this issue is not yet visible on https://www.githubstatus.com/. I'm afraid that GitHub is not aware of it. Is there another way to alert GitHub?

**Comment by kpaaro** [comment]
For us the issue got reolved a few hour ago, and for the mean while I changed my hosts file to direct github.com to the US region load balancer.
The IP was provided in another thread about this issue.

**Reply by kpaaro** [reply]
![image](https://github.com/community/community/assets/2753157/ad700169-9618-4171-9520-2ed5c213f145)

**Reply by marcolopes** [reply]
Seems ok now... (yet a bit slower than usual)

**Comment by cdf1982** [comment]
Hello, in the last week I've also experienced frequent "Failed to connect to github.com port 443: Connection timed out" messages from different repositories; oddly enough, switching the IP address I connect from via VPN seems to solve. Are there issues with a too-aggressive DDOS prevention mechanism?

**Reply by kpaaro** [reply]
Last week et seemed more of a loadbalancing/dns issue, where one or more 'dead' ip's were still given out. And switching to the US ip solved the issue then. By now we have set everything back to normal.

**Comment by martin-coded** [comment]
I have this issue since today. Are there still problems?

**Reply by marcolopes** [reply]
All is fine here... (Slow, as usual, but no problems)

**Reply by martin-coded** [reply]
Thank you for confirmation! I just figured out, that a firewall was blocking the request for some unknown reason.

**Comment by marcolopes** [comment]
Problems are back... can't do anything!

**GitHub Desktop Acess**:
fatal: unable to access 'https://github.com/marcolopes/dma.git/': Failed to connect to github.com port 443 after 21246 ms: Couldn't connect to server

**SVN Access**:
org.apache.subversion.javahl.ClientException: Can't create session
svn: Commit failed (details follow):
svn: Unable to connect to a repository at URL 
Connection timed out
svn: Connection timed out

**Comment by xingshengao** [comment]
Close fireWall software, and sloved.

**Reply by ludlows** [reply]
thanks. your suggestion solved my problem

**Reply by dongs95** [reply]
thank you，this is right on windows

**Reply by kranthivardhan** [reply]
thanks worked

**Reply by pkq11** [reply]
thanks, works on windows

**Reply by Adrian1jr** [reply]
how you close that? I already disabled my firewall on windows 11 but is not working, any help?

**Reply by Peter-Pan010612** [reply]
you can try restart it

**Reply by Adrian1jr** [reply]
I solved my problem by using a USB device to get Wifi on my pc and connecting to “other” wifi, basically the same network but in wifi mode lol

**Reply by leekinm** [reply]
Actually, closing firewall works well, but if i keep truning the firewall off whether my computer will be attacked or not? Is there any permant solution?

**Reply by studyingforever1** [reply]
yes i solve my problem by closing firewall too thanks . by the way, the guy upstrair, you can close your firewall after push your code.

**Reply by journey0616** [reply]
> you can try restart it

nice! restart computer perfectly to solve my problem

**Comment by staramidst** [comment]
I just made an ssh key with ssh-keygen put the key in the folder I made and used it to add my key and then tested it with git clone ssh the repo copy and paste.

**Reply by Stephanommg** [reply]
This worked for me

**Comment by ZupoLlask** [comment]
A few months ago, while certain operations (like git clone) still were working fine, some others (like git checkout) started to behave as described by you guys.

I've just find out that the culprit was using the URL with ".git" suffix (e.g., https://github.com/marcolopes/dma.git instead of simply https://github.com/marcolopes/dma).

As soon as I removed that ".git" suffix from my URLs, all the trouble went away.

**Reply by marcolopes** [reply]
Thanks! This does not matter anymore because GIT has dropped SVN support... (and the problems were related to SVN use)

**Comment by boxingnguyen** [comment]
Same issue from Vietnam, solved by connect from via VPN cloudflare 1.1.1.1 =))

**Comment by macrozone** [comment]
same issue here all of sudden (switzerland)

**Comment by ezequielabregu** [comment]
Same issue here in Germany!  :-(

**Comment by bifacil** [comment]
Same here since yesterday (spain)

**Comment by chithambaramoorthii** [comment]
Same issue here from (India)

**Reply by ghost** [reply]
you find solution of these issue

**Comment by xEverth** [comment]
Action failed because of this during auto-commit step

**Comment by Danthesage01** [comment]
Just experienced this issue now (Nigeria)

**Comment by dev-zeb** [comment]
Same issue here, facing since this morning. (Bangladesh)
However, when I use mobile data using Hotspot the issue doesn't occur. It's occurring only when I'm connected to the Wi-Fi.

**Comment by Ardiannn08** [comment]
same issue here from (Indonesia) since last 3 days

**Comment by shadan1234** [comment]
I have found the solution to the problem !!! . Just switch your wifi , then run the same git cmds , and it will work.

**Comment by lincolncesardev** [comment]
there was a vs.code folder containing a file called .settings . there was a port on default. Deleted and the clone was ok here

**Comment by jenniebxiao** [comment]
same issue here
`
Failed to connect to github.com port 443 after 21045 ms: Couldn't connect to server
`

I changed my network, and issue fixed

**Comment by LeftoverAtoms** [comment]
```
`git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin` exited with an unexpected code: 128.
stderr:
fatal: unable to access 'repository': Failed to connect to github.com port 443 after 21107 ms: Couldn't connect to server
```

Had the same issue, toggling Wifi off/on solved the issue.

**Reply by chanmaoganda** [reply]
Same issue, toggling Wifi works!

**Comment by PiyushBagde** [comment]
i am experiencing the same, does anyone know how to fix `this?`
```
fatal: unable to access (repo link) ': 
Failed to connect to github.com port 443 after 21168 ms: Couldn't connect to server
```

**Reply by PiyushBagde** [reply]
i figured out something, if u make pull / push multiple times like 10 times or something, it works at least one time.

**Reply by ritiek** [reply]
Hi, I'm curious if this helps you:
https://github.com/orgs/community/discussions/29269#discussioncomment-10094288

**Reply by PiyushBagde** [reply]
> Hi, I'm curious if this helps you: https://github.com/orgs/community/discussions/29269#discussioncomment-10094288

Thanks

**Reply by viethung0823** [reply]
try to do it multiple times works but it's so annoy when we do not know the root caused

**Reply by chanmaoganda** [reply]
perhaps set your dns server to google server?(8.8.8.8)

**Reply by PiyushBagde** [reply]
> try to do it multiple times works but it's so annoy when we do not know the root cause

Turn off private DNS if you using one

**Reply by viethung0823** [reply]
> > try to do it multiple times works but it's so annoy when we do not know the root cause
> 
> Turn off private DNS if you using one

I've had checked network settings, no dns, no firewall, or VPN was on.

**Comment by gertcuykens** [comment]
Same issue. ipv6 related maybe ? I don't have ipv4 address

**Comment by PalashCoder** [comment]
Same issue I've faced.
This issue can be occurred when you previously pulled from the same repo and forget to run "npm i" this will install the dependencies according to the package.json and after running npm i, you can push your code without any problem.

**Comment by upsampling** [comment]
Same issue here in Mexico.

curl: (28) Failed to connect to github.com port 443 after 138319 ms: Couldn't connect to server

**Reply by TerenceTian625** [reply]
Are you using aws ec2 instance as a self-hosted runner, I'm facing the same error as yours when spinning up ec2 instance as self-hosted runner. Sometime it worked but sometimes it failed.

**Reply by franklizh** [reply]
here my steps to solve this problem:
find your computer internet  setting ,find proxy,address,port
then input code in git bash:
```
git config --global http.proxy http://127.0.0.1:your port num
git config --global https.proxy http://127.0.0.1:your port num
```

**Reply by Morning-Ye** [reply]
This works for me, thanks!

**Comment by franklizh** [comment]
here my steps to solve this problem:
find your computer internet  setting ,find proxy,address,port
then input code in git bash:
```
git config --global http.proxy http://127.0.0.1:your port num
git config --global https.proxy http://127.0.0.1:your port num
```

**Reply by liangmouse** [reply]
I resolved it by toggling Wifi off/on.

**Reply by Harbinder04** [reply]
It works or you can also try switching on/off airplane mode.

**Reply by yayapi201** [reply]
> here my steps to solve this problem: find your computer internet setting ,find proxy,address,port then input code in git bash:
> 
> ```
> git config --global http.proxy http://127.0.0.1:your port num
> git config --global https.proxy http://127.0.0.1:your port num
> ```

works for me

**Reply by zxq-clt** [reply]
very good

**Reply by heigguys** [reply]
its cool

**Reply by SYChen123** [reply]
it works for me

**Reply by DingDingDag** [reply]
Thank you, it work for me

**Comment by JustCod101** [comment]
I think i have a good solution for you , JUST turn on **TUN MODE** in your proxy software

**Reply by wusctong** [reply]
**IT WORKS!!!!!!!!**

**Reply by BlackStar1453** [reply]
It works for me too.

**Reply by 0xbul1** [reply]
wow，this way works for me

**Reply by Harryking1999** [reply]
it really works

**Reply by David3707** [reply]
Thanks, it helped.

**Reply by ZachRode** [reply]
正解

**Reply by bod250** [reply]
danm right

**Comment by FitzHoo** [comment, marked as answer]
I also tried a lot of methods mentioned above, but none of them fixed my problem.
Finally, I find the following method useful for me and hope it will help you as well.

**[Use SSH instead of HTTPS]**
If you have problems with HTTPS access, you can try pushing to GitHub using the SSH protocol.
1. You need to generate an SSH key and add it to your GitHub account.
2. Add an SSH key to GitHub:
    Copy the generated public key (~/.ssh/id_rsa.pub).
    Log in to GitHub and go to Settings -> SSH and GPG keys -> New SSH keys.
3. Change the remote warehouse URL to SSH format:
   git remote set-url origin git@github.com:{repo_name}/{name}.git

**Reply by luchaoqun123** [reply]
牛逼

**Reply by D1scowolf** [reply]
感谢！这边补充两点：
1. 在我这边首先需要使用备用端口（似乎是网络屏蔽了22端口）。在你的.ssh/config中添加：
  Host github.com
  Hostname ssh.github.com
  Port 443
  User git
2. 这里的repo_name似乎是组织名或用户名，name才是仓库名？4o是这么说的，我实际这么做也过了

**Reply by 1985836841** [reply]
666

**Reply by shandonguzi** [reply]
<img width="404" height="339" alt="Snipaste_2025-08-07_10-59-32" src="https://github.com/user-attachments/assets/541307a1-ea21-4f9d-aaa5-ff97af7dcf37" />

**Comment by yayapi201** [comment]
good to know! Are you a AI user or researcher? I am an architectural designer, trying to learn by myself



---- Replied Message ----
| From | ***@***.***> |
| Date | 03/29/2025 09:09 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [community/community] Failed to connect to github.com port 443 after 21124 ms: Couldn't connect to server (Discussion #66424) |

It works for me too.

**Comment by paulHYF** [comment]
git config --global http.postBuffer 524288000

**Reply by chuanminglu** [reply]
正解

**Reply by Chord-Chen-30** [reply]
It worked for me!

**Reply by RandomGuy1520** [reply]
worked for me :)

**Reply by simonhou** [reply]
It worked for me too!

**Reply by celery8911** [reply]
It worked for me too!

**Reply by amntheap** [reply]
worked for me too!

**Reply by jsdhiwhfd** [reply]
worked for me :)thk a lot dude

**Reply by Green-404** [reply]
worked for me! Thanks!

**Reply by wulanx1324** [reply]
It worked for me too!

**Reply by hwhsu1231** [reply]
It worked for me too!

**Comment by Shubham2816** [comment]
I encountered similar issues and attempted various solutions without success. However, the following method proved effective in my case, and I hope it will be helpful for others facing the same problem.

✅ Use SSH Instead of HTTPS
If you're experiencing issues with HTTPS access when working with GitHub, switching to the SSH protocol for pushing changes might resolve the problem.

🔐 Step 1: Generate an SSH Key
If you haven't already, generate an SSH key on your local machine using the following command:

ssh-keygen -t rsa -b 4096 -C "your_email@example.com"

This will generate a key pair, typically located at ~/.ssh/id_rsa (private) and ~/.ssh/id_rsa.pub (public).

🔗 Step 2: Add SSH Key to Your GitHub Account
Copy the contents of the public key file:

cat ~/.ssh/id_rsa.pub

Log in to your GitHub account.

Navigate to Settings → SSH and GPG keys → New SSH key.

Paste the copied public key, give it a recognizable title, and save.

🔄 Step 3: Update the Git Remote URL
Replace the HTTPS URL with the SSH format using the command below:

git remote set-url origin git@github.com:USERNAME/REPOSITORY.git

Replace USERNAME and REPOSITORY with your actual GitHub username and repository name.

Once completed, you should be able to push and pull from your GitHub repository using SSH without further issues.

**Comment by fhfeishi** [comment]
git config --global url."git@github.com:".insteadOf "https://github.com/"

**Reply by celery8911** [reply]
It worked for me.

**Reply by daliqinchuanniu** [reply]
me too

**Comment by axai-kaizoku** [comment]
Might be an internet issue !

**Comment by narendermann** [comment]
Try checking routes on the machine. I got the same error and later realised misconfigured routes. I fixed the routes and worked very well.

**Reply by PhilippeCarphin** [reply]
What do you mean by fixing the routes?

**Comment by NTHoang2103** [comment]
1. Test Network Connectivity
First, confirm that your machine can reach github.com at all.

Test with ping (Basic):

Bash

ping github.com
Expected Result: You should see successful replies with low latency (e.g., <100ms).

If it fails: Your basic DNS or network connection is broken.

Test with telnet or nc (Port-specific): This tests the exact port Git uses for HTTPS.

Bash

# On Linux/macOS
nc -vz github.com 443

# On Windows (using telnet, might need to be enabled)
telnet github.com 443
Expected Result: A message indicating the connection succeeded (e.g., succeeded! or an empty terminal).

If it fails: This strongly suggests a firewall or proxy is blocking port 443 traffic to GitHub.

2. Check for Firewall or Antivirus Blocks
Your local machine's firewall or security software is a very common culprit for blocking outbound connections.

Local Firewall: Check your operating system's firewall (Windows Defender Firewall, macOS Firewall, or Linux's ufw/iptables) settings. Ensure that Git and/or your terminal/shell application are allowed to make outbound connections on port 443.

Antivirus/Security Software: Temporarily disable your antivirus or VPN/security suite and try the Git command again. If it works, you've found the issue and need to configure an exception for Git.

3. Check Proxy Settings
If you are on a corporate network, a school network, or using a VPN, you likely need to configure Git to use a proxy server.

Configure Git to use a Proxy (HTTP/HTTPS):

Bash

# Set the proxy globally
git config --global http.proxy http://user:password@proxy.example.com:8080

# Check the setting
git config --global http.proxy
Replace the example address and port with your actual proxy details.

4. Switch to SSH (Alternative Connection Method)
If the HTTPS connection (port 443) continues to fail, you can switch your remote URL to use the SSH protocol, which uses port 22.

Prerequisites: You must have an SSH key configured and added to your GitHub account.

Change Remote URL:

Get the SSH URL for your repository (it looks like git@github.com:username/repo.git).

In your local repository, run:

Bash

git remote set-url origin git@github.com:username/repo.git
Try pushing/pulling again:

Bash

git pull
If SSH works, it bypasses the Port 443 issue entirely.

5. Clear DNS Cache and Reset TCP/IP
If you suspect a local network or DNS issue, try clearing your caches:

Windows:

Bash

ipconfig /flushdns
netsh winsock reset
netsh int ip reset
macOS/Linux:

Bash

sudo dscacheutil -flushcache  # macOS
sudo killall -HUP mDNSResponder # macOS
(Note: Linux DNS clearing varies by distribution/setup)

After running the commands, restart your computer to ensure the network stack is fully reset.

**Comment by Chuangye-Wang** [comment]
I solved this through adding a .wslconfig file under C:/Users/username.
